ids/id.go

```diff
@@ -32,11 +32,39 @@ var (
 // ID wraps a 32 byte hash used as an identifier
 type ID [IDLen]byte
 
+// Writeable is an helper which helps modifiying ID content
+func Writable(id *ID) []byte {
+	return id[:]
+}
+
+// Any modification to Bytes will be lost since id is passed-by-value
+// Use Writable(ID) is you need to modify the ID
+func (id ID) Bytes() []byte {
+	return id[:]
+}
+
 // ToID attempt to convert a byte slice into an id
 func ToID(bytes []byte) (ID, error) {
 	return hashing.ToHash256(bytes)
 }
 
+// UnsafeToID attempt to convert a byte slice into an id
+// It's up to caller to make sure bytes are well formatted.
+// Should be used in tests only.
+func UnsafeToID(bytes []byte) ID {
+	rightLenSlice := make([]byte, IDLen)
+	copy(rightLenSlice, bytes)
+	res, _ := hashing.ToHash256(rightLenSlice)
+	return res
+}
+
+func (id ID) String() string {
+	// We assume that the maximum size of a byte slice that
+	// can be stringified is at least the length of an ID
+	s, _ := cb58.Encode(id[:])
+	return s
+}
```

Comment:
nit: undo diff (I know you're moving code to be in more expected spots/near common code, but since we're introducing new code in this PR let's try to avoid cluttering the diff w/ refactoring code)

Reply:
Done